cmd/maya-apiserver/spc-actions/select_disk.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2017 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storagepoolactions
+
+import (
+	mach_apis_meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//openebs "github.com/openebs/maya/pkg/client/clientset/versioned"
+	"errors"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	openebs "github.com/openebs/maya/pkg/client/generated/clientset/internalclientset"
+	"github.com/openebs/maya/pkg/client/k8s"
+)
+
+// clientset struct holds the interface of internalclientset
+// i.e. openebs.
+// This struct will be binded to method ListDisk and is useful in mocking
+// and unit testing.
+type clientSet struct {
+	oecs openebs.Interface
+}
+
+// nodeDisk struct will be used as a value for a map nodeDiskMap (map defined in ListDisk function)
+// The struct will be useful in forming the data structure nodeDiskMap which will be manipulated
+// to efficiently select the nodes and disk for dynamic pool provisioning.
+type nodeDisk struct {
+	// diskCount is the count of usable disks that can be used in storagepool provisioning.
+	diskCount int
+	//diskList is the list of usable disks that can be used in storagepool provisioning.
+	diskList []string
+}
+
+// getDiskList is a wrapper function which will receive list of disks
+// that can be used for dynamic pool provisioning at runtime.
+// The function finally returns the disk list to the caller (i.e. getCasPoolDisk function).
+func getDiskList(cp *v1alpha1.CasPool) ([]string, error) {
+
+	// Get kubernetes clientset
+	// namespaces is not required, hence passed empty.
+	newK8sClient, err := k8s.NewK8sClient("")
+
+	if err != nil {
+		return nil, err
+	}
+	// Get openebs clientset using a getter method (i.e. GetOECS() ) as
+	// the openebs clientset is not exported.
+	newOecsClient := newK8sClient.GetOECS()
+
+	// Create instance of clientset struct defined above which binds
+	// ListDisk method and fill it with openebs clienset (i.e.newOecsClient ).
+	newClientSet := clientSet{
+		oecs: newOecsClient,
+	}
+
+	// if no minimum pools were specified it will default to 1.
+	if cp.MinPools <= 0 {
```

Comment:
This validation should ideally not happen in this function.

Reply:
Sure... will correct this.